src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -72,28 +75,31 @@ public function __construct(array $configuration, array $connectionCredentials =
         }
 
         $initializer = static function ($redis) use ($host, $port, $auth, $serializer, $dbIndex) {
-            $redis->connect($host, $port);
+            if (null === $redis) {
+                $redis = !\is_array($host) ? new \Redis() : new \RedisCluster(null, $host);
+            }
+
+            if ($redis instanceof \Redis) {
+                $redis->connect($host, $port);
+            }
+
             $redis->setOption(\Redis::OPT_SERIALIZER, $serializer);
 
-            if (null !== $auth && !$redis->auth($auth)) {
+            if (null !== $auth && $redis instanceof \Redis && !$redis->auth($auth)) {
```

Comment:
what about auth for RedisCluster ?

Reply:
It's not supported by `RedisCluster`, you get `Call to undefined method RedisCluster::auth()` if you try use this method.

Reply:
Fixed:

https://github.com/symfony/symfony/blob/8d6de3a91e24ff63472955a45be7f9dc92a031a3/src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php#L77-L82